src/CommitmentsV2/SFA.DAS.CommitmentsV2/Models/Commitment.cs

```diff
@@ -57,40 +57,40 @@ public virtual void AddDraftApprenticeship(DraftApprenticeshipDetails draftAppre
         private void ValidateDraftApprenticeshipDetails(DraftApprenticeshipDetails draftApprenticeshipDetails, IUlnValidator ulnValidator)
         {
             var errors = new List<DomainError>();
-            errors.AddRange(BuildNameValidationFailures(draftApprenticeshipDetails));
+            errors.AddRange(BuildFirstNameValidationFailures(draftApprenticeshipDetails));
+            errors.AddRange(BuildLastNameValidationFailures(draftApprenticeshipDetails));
             errors.AddRange(BuildEndDateValidationFailures(draftApprenticeshipDetails));
             errors.AddRange(BuildCostValidationFailures(draftApprenticeshipDetails));
             errors.AddRange(BuildReferenceValidationFailures(draftApprenticeshipDetails));
-                
+            errors.AddRange(BuildUlnValidationFailures(draftApprenticeshipDetails, ulnValidator));
             errors.ThrowIfAny();
         }
 
-        private IEnumerable<DomainError> BuildNameValidationFailures(DraftApprenticeshipDetails draftApprenticeshipDetails)
+        private IEnumerable<DomainError> BuildFirstNameValidationFailures(DraftApprenticeshipDetails draftApprenticeshipDetails)
         {
-
             if (string.IsNullOrWhiteSpace(draftApprenticeshipDetails.FirstName))
             {
                 yield return new DomainError(nameof(draftApprenticeshipDetails.FirstName), "First name must be entered");
+                yield break;
             }
-            else
-            {
-                if (draftApprenticeshipDetails.FirstName.Length > 100)
-                {
-                    yield return new DomainError(nameof(draftApprenticeshipDetails.FirstName), "You must enter a first name that's no longer than 100 characters");
 
-                }
+            if (draftApprenticeshipDetails.FirstName.Length > 100)
```

Comment:
This will fail when the first name isn't entered. Be better with a "else if()" construct

Reply:
the earlier block would catch this and break